src/services/accounts/AccountsConvertService.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { Keyring } from '@polkadot/api';
+import { isHex } from '@polkadot/util';
+import { allNetworks } from '@polkadot/util-crypto';
+import { blake2AsHex } from '@polkadot/util-crypto';
+import { BadRequest } from 'http-errors';
+
+import { IAccountConvert } from '../../types/responses/AccountConvert';
+import { AbstractService } from '../AbstractService';
+
+export class AccountsConvertService extends AbstractService {
+	/**
+	 * Takes a given accountId and converts it to a SS58 address
+	 * based on the values of the variables scheme, ss58Prefix & publicKey.
+	 * It also returns the network name.
+	 *
+	 * @param address ss58 or hex address to validate
+	 * @param scheme
+	 * @param ss58Prefix
+	 * @param publicKey
+	 */
+	accountConvert(
+		accountId: string,
+		scheme: 'ed25519' | 'sr25519' | 'ecdsa',
+		ss58Prefix: number,
+		publicKey: boolean
+	): IAccountConvert {
+		const accountIdIsHex = isHex(accountId);
+		if (!accountIdIsHex) {
+			throw new BadRequest('`accountId` is not a valid hex value.');
```

Comment:
When we log this value as an Error is this referring to the `address` param in the URI.

Reply:
Yes it is referring to the path parameter AccountId or Public Key (hex) that should be converted to an SS58 address.

Reply:
Do you think we should write `The given path parameter "accountId" is not a valid hex value`?

Reply:
After your first suggestion I changed most of the error messages included this one. 
It is now 
`'The `accountId` parameter provided is not a valid hex value.' `

I also like the one you suggested. Whatever you think is best.